scripts/task.ts

```diff
@@ -31,166 +58,407 @@ export type TemplateDetails = {
 type MaybePromise<T> = T | Promise<T>;
 
 export type Task = {
+  /**
+   * A description of the task for a prompt
+   */
+  description: string;
+  /**
+   * Does this task represent a service for another task?
+   *
+   * Unlink other tasks, if a service is not ready, it doesn't mean the subsequent tasks
+   * must be out of date. As such, services will never be reset back to, although they
+   * will be started if dependent tasks are.
+   */
+  service?: boolean;
   /**
    * Which tasks run before this task
    */
-  before?: TaskKey[];
+  before?: TaskKey[] | ((options: PassedOptionValues) => TaskKey[]);
```

Comment:
the `before` key in confused me a bit when I saw it in a task, because I thought it meant "this task must go before this other task", but seeing these comments tells me it's the other way around.

In GitHub Actions it's called "needs" and nx + Turborepo calls it "dependsOn", which I think are both more descriptive names, that we could also use.

Reply:
Let's go with `dependsOn`